new-components/reporters/pdf/internal/reporter/reporter.go

```diff
@@ -0,0 +1,204 @@
+package reporter
+
+import (
+	"context"
+	_ "embed"
+	"fmt"
+	"html/template"
+	"log/slog"
+	"os"
+	"path/filepath"
+	"time"
+
+	"github.com/go-errors/errors"
+	"github.com/smithy-security/pkg/env"
+
+	playwright "github.com/smithy-security/smithy/pkg/playwright"
+	s3client "github.com/smithy-security/smithy/pkg/s3"
+	"github.com/smithy-security/smithy/sdk/component"
+	vf "github.com/smithy-security/smithy/sdk/component/vulnerability-finding"
+)
+
+// NewReporter returns a new PDF reporter.
+func NewReporter(conf *Conf) *PdfReporter {
+	return &PdfReporter{
+		conf: conf,
+	}
+}
+
+type PdfReporter struct {
+	conf *Conf
+}
+
+type (
+	Conf struct {
+		Bucket       string
+		Region       string
+		SkipS3Upload bool
+	}
+)
+
+// NewConf returns a new configuration build from environment lookup.
+func NewConf(envLoader env.Loader) (*Conf, error) {
+	var envOpts = make([]env.ParseOption, 0)
+	if envLoader != nil {
+		envOpts = append(envOpts, env.WithLoader(envLoader))
+	}
+
+	skipS3Upload, err := env.GetOrDefault(
+		"SKIP_S3_UPLOAD",
+		true,
+		append(envOpts, env.WithDefaultOnError(false))...,
+	)
+	if err != nil {
+		return nil, errors.Errorf("could not get env variable for SKIP_S3_UPLOAD: %w", err)
+	}
+
+	bucket, err := env.GetOrDefault(
+		"BUCKET_NAME",
+		"",
+		append(envOpts, env.WithDefaultOnError(false))...,
+	)
+	if err != nil {
+		return nil, errors.Errorf("could not get env variable for BUCKET_NAME: %w", err)
+	}
+
+	region, err := env.GetOrDefault(
+		"BUCKET_REGION",
+		"",
+		append(envOpts, env.WithDefaultOnError(false))...,
+	)
+	if err != nil {
+		return nil, errors.Errorf("could not get env variable for BUCKET_REGION: %w", err)
+	}
+
+	return &Conf{
+		Bucket:       bucket,
+		Region:       region,
+		SkipS3Upload: skipS3Upload,
+	}, nil
+}
+
+func (p PdfReporter) Report(
+	ctx context.Context,
+	findings []*vf.VulnerabilityFinding,
+) error {
+	logger := component.LoggerFromContext(ctx)
```

Comment:
I think that it would be nice to add a field to this logger to check for S3 Upload being enabled/disabled :)

`logger := component.LoggerFromContext(ctx).With(slog.Bool("s3_upload_enabled", p.conf.SkipS3Upload))`